sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -211,6 +211,40 @@ object GpuOrcScan extends Arm {
         } else {
           downCastAnyInteger(col, toDt)
         }
+      // bool to float, double(float64)
+      case (DType.BOOL8, DType.FLOAT32 | DType.FLOAT64) =>
+        col.castTo(toDt)
+      // bool to string
+      case (DType.BOOL8, DType.STRING) =>
+        withResource(col.castTo(toDt)) { casted =>
+          // cuDF produces "ture"/"false" while CPU outputs "TRUE"/"FALSE".
+          casted.upper()
+        }
+      // integer to float, double(float64), string
+      case (DType.INT8 | DType.INT16 | DType.INT32 | DType.INT64,
+      DType.FLOAT32 | DType.FLOAT64 | DType.STRING) =>
+        col.castTo(toDt)
+      // {bool, integer} to timestamp(micro seconds)
+      case (DType.BOOL8 | DType.INT8 | DType.INT16 | DType.INT32, DType.TIMESTAMP_MICROSECONDS) =>
+        // cuDF requires casting to a long first.
+        withResource(col.castTo(DType.INT64)) { longs =>
+          // In CPU, ORC assumes the integer value is in seconds, and returns timestamp in
+          // micro seconds.
+          withResource(longs.bitCastTo(DType.TIMESTAMP_SECONDS)) { seconds =>
+            seconds.castTo(DType.TIMESTAMP_MICROSECONDS)
+          }
+        }
+      case (DType.INT64, DType.TIMESTAMP_MICROSECONDS) =>
+        // We need overflow checking here, since max value of INT64 is about 9 * 1e18, and convert
+        // INT64 to micro seconds(also a INT64 actually), we need multiply 1e6, it may cause long
+        // overflow.
+        // If val * 1e6 / 1e6 == val, then there is no overflow.
+        if (!canMultiply(col, 1e6.toLong)) {
+          throw new ArithmeticException("Long overflow")
```

Comment:
Please use the `RapidsErrorUtil.overflowException`, so that we could match the error type with Spark's.

Reply:
Fixed. Many thx.